bundles/mapping/mapmodule/MapModuleClass.ol.js

```diff
@@ -538,24 +599,39 @@ export class MapModule extends AbstractMapModule {
      * @param {Boolean} suppressEnd true to NOT send an event about the map move
      *  (other components wont know that the map has moved, only use when chaining moves and
      *     wanting to notify at end of the chain for performance reasons or similar) (optional)
+     * @param {Object} options animation name as string
+     *     Usable animations: fly/pan/zoomPan
      * @return {Boolean} success
      */
-    centerMap (lonlat, zoom, suppressEnd) {
+    centerMap (lonlat, zoom, suppressEnd, options) {
+        const view = this.getMap().getView();
         lonlat = this.normalizeLonLat(lonlat);
         if (!this.isValidLonLat(lonlat.lon, lonlat.lat)) {
             return false;
         }
-        this.getMap().getView().setCenter([lonlat.lon, lonlat.lat]);
         if (zoom === null || zoom === undefined) {
             zoom = this.getMapZoom();
         }
-        this.getMap().getView().setZoom(zoom);
+
+        const zoomN = zoom.type === 'scale' ? view.getZoomForResolution(zoom.value) : zoom.value;
+        if (typeof options === 'undefined') {
+            view.animate({
+                center: [lonlat.lon, lonlat.lat],
+                zoom: zoomN,
+                duration: 3000,
+                easing: easeOut
+            });
+        } else {
+            this._animateTo(lonlat, zoomN, options.animation, options.duration);
+        }
+
         this.updateDomain();
         if (suppressEnd !== true) {
             this.notifyMoveEnd();
```

Comment:
With this implementation the map moved event comes at the beginning of the animation. We would need to trigger it at the end.

Reply:
Also I think we need to have centerMap() to skip any animation by default as users might expect the map to be on a given location after this is called.